Improve Import/Export Workflow with User Options and VMD Bug Fixes

rintrint opened the pull request:
- [VMD] Add "Preserve Animation Curves" option
- [VMD] Add User-Controlled Camera Change Detection Option
- [VMD] Add User-Controlled Light Change Detection Option
- [VMD] Add Support for Starting Import at Current Frame Position
- [VMD] Add "Always Create New Action" option
- [VMD] Add VMD Exporter Test Script and Fix Related Issues
- [Docs] Update CHANGELOG, CONTRIBUTORS, and README
- [Prefs] Add Option for Auto-Applying Default Preset (zero-click setup)
- [Code] Improve Code Quality

Related Issues:
- #183
- #216
- #219
- #221

nagadomi commented on mmd_tools/auto_scene_setup.py:
```diff
@@ -6,10 +6,16 @@
-        ts, te = i.frame_range
-        s = min(s, ts)
-        e = max(e, te)
+    for action in bpy.data.actions:
+        # Blend frame ranges from two imported VMD files
+        # Get first imported VMD range
+        action.use_frame_range = True
+        ts, te = action.frame_range
+        s, e = min(s, ts), max(e, te)
+        # Get second imported VMD range
+        action.use_frame_range = False
+        ts, te = action.frame_range
+        s, e = min(s, ts), max(e, te)
```
If the intention is not to set `action.use_frame_range = False` for all existing actions, it would be safer to restore their original values.

nagadomi commented on mmd_tools/core/vmd/exporter.py:
```diff
@@ -40,18 +44,18 @@ def frameNumbers(self):
+            if self.__preserve_curves and kp0.interpolation != "LINEAR" and kp1.co.x - kp0.co.x > 2.5:
                 if kp0.interpolation == "CONSTANT":
-                    result.add(int(kp1.co[0] - 0.5))
+                    result.add(math.floor(kp1.co[0]))
```
This line seems to be trying to reproduce CONSTANT interpolation with 1-interval keyframe (for motion created on blender), so I think it must be one frame before `round(kp1.co[0])`.
It should also not be -1 when the start frame is 0.
It may not be simple to write because `int()` and `round()` have different rounding results for negative values.

rintrint replied:
fixed

nagadomi commented:
>I also want to disable the "Rename Bones - L / R Suffix" option by default when importing PMX files.

L/R naming is a feature almost always used when modeling in Blender.
If this naming rule is not followed, many basic Blender features won't work.
So I think it's a highly recommended option.

rintrint commented:
I've reverted the **"Rename Bones - L / R Suffix"** option default back to `True` and added a clear description for it. Also fixed the issues found by Copilot.

Thanks!
